### PR TITLE
feat: doctoc

### LINKED ISSUE #20: Add `doctoc` back
## Overview

We removed `doctoc` in #16 in favor of `mdbook-toc`, but some folks prefer viewing the specs on GitHub. Migrate back to `doctoc` and remove the `mdbook-toc` plugin.

### PR DESCRIPTION
**Pull request by tynes**
**Description**

It was decided that we would merge the mdbook PR even though it removed doctoc. This PR follows up to add doctoc back. This allows the table of contents to be viewed on github without needing to rely completely on mdbook

### CODE REVIEW

File: specs/protocol/withdrawals.md [hunk 1]
```diff
@@ -10,6 +29,8 @@
 [g-relayer]: ../glossary.md#withdrawals
 [g-execution-engine]: ../glossary.md#execution-engine
 
+## Overview
+
```
Comment (clabby) [inline specs/protocol/withdrawals.md:33]: There is a `<!-- toc -->` tag below this with another `**Table of Contents**` header, can we remove it?

**Review by clabby — CHANGES_REQUESTED** [review]
LGTM; Thanks for doing this. Will approve once `withdrawals.md` has been fixed up.

File: specs/protocol/withdrawals.md (hunk 1 above)
Comment (tynes) [reply specs/protocol/withdrawals.md:33]: Yes sorry I thought I got all of them

File: README.md [hunk 2]
```diff
@@ -30,5 +30,5 @@ cargo install mdbook mdbook-katex mdbook-toc mdbook-linkcheck mdbook-mermaid
 ### Serving the book locally
 
 ```sh
```
Comment (clabby) [inline README.md:32]: Can we remove the direction to install `mdbook-toc` below? No longer needed.

**Comment by clabby** [comment]
We also still install `mdbook-toc` in the deploy workflow, no longer needed.

**Comment by clabby** [comment]
Same with `book.toml` - can remove this section:

```toml
[preprocessor.toc]
command = "mdbook-toc"
renderer = ["html"]
```

File: README.md (hunk 2 above)
Comment (clabby) [reply README.md:32]: Also would be great to add a section reminding folks to run `just lint-specs-toc-check` before submitting PRs. I removed this section from `linting.md` too, will need to be added back.

**Review by clabby — APPROVED** [review]

**Review by mslipper — APPROVED** [review]

**Pull request merged**